src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -2,147 +2,161 @@ import "@lit-labs/virtualizer";
+
   @state() private _narrow = false;
 
   @state() private _hint?: string;
 
+  @state() private _entities?: EntityRegistryEntry[];
+
+  @state() private _areas?: AreaRegistryEntry[];
+
+  @state() private _devices?: DeviceRegistryEntry[];
+
   @query("ha-textfield", false) private _filterInputField?: HTMLElement;
 
+  // @ts-ignore
+  @LocalStorage("suggestions", true, {
+    attribute: false,
+  })
+  private _suggestions: QuickBarItem[] = [];
+
   private _focusSet = false;
 
   private _focusListElement?: ListItem | null;
 
+  private _filterItems = memoizeOne(
+    (items: QuickBarItem[], filter: string): QuickBarItem[] =>
+      fuzzyFilterSort<QuickBarItem>(filter.trimLeft(), items)
+  );
+
   public async showDialog(params: QuickBarParams) {
-    this._commandMode = params.commandMode || this._toggleIfAlreadyOpened();
     this._hint = params.hint;
     this._narrow = matchMedia(
       "all and (max-width: 450px), all and (max-height: 500px)"
     ).matches;
-    this._initializeItemsIfNeeded();
-    this._open = true;
+  }
+
+  public hassSubscribe(): UnsubscribeFunc[] {
+    return [
+      subscribeAreaRegistry(this.hass.connection!, (areas) => {
+        this._areas = areas;
+      }),
+      subscribeDeviceRegistry(this.hass.connection!, (devices) => {
+        this._devices = devices;
+      }),
+      subscribeEntityRegistry(this.hass.connection!, (entities) => {
+        this._entities = entities;
+      }),
+    ];
   }
 
   public closeDialog() {
-    this._open = false;
     this._opened = false;
     this._focusSet = false;
     this._filter = "";
     this._search = "";
     fireEvent(this, "dialog-closed", { dialog: this.localName });
   }
 
-  private _getItems = memoizeOne(
-    (commandMode: boolean, commandItems, entityItems, filter: string) => {
-      const items = commandMode ? commandItems : entityItems;
-
-      if (items && filter && filter !== " ") {
-        return this._filterItems(items, filter);
-      }
-      return items;
+  protected willUpdate(changedProperties: PropertyValues): void {
+    super.willUpdate(changedProperties);
+    if (
+      (changedProperties.has("_entity") ||
+        changedProperties.has("_areas") ||
+        changedProperties.has("_devices")) &&
+      this._areas &&
+      this._devices &&
+      this._entities
+    ) {
+      this._initializeItems();
+      this._opened = true;
     }
-  );
+  }
 
-  protected render() {
-    if (!this._open) {
+  protected render(): TemplateResult {
+    if (!this._opened) {
       return html``;
     }
 
-    const items: QuickBarItem[] | undefined = this._getItems(
-      this._commandMode,
-      this._commandItems,
-      this._entityItems,
-      this._filter
-    );
+    let sectionCount = 0;
+
+    if (this._items && this._filter && this._filter !== "") {
+      const newFilteredItems: QuickBarItem[] = [];
+      this._items.forEach((arr) => {
+        const items = this._filterItems(arr, this._filter).slice(0, 3);
```

Comment:
Why do we show only 3 items per section? Searching an entity by domain will be very hard now?